atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -279,6 +292,19 @@ Stream<BatchingVisitable<RowResult<byte[]>>> getRangesLazy(
     @Idempotent
     void delete(TableReference tableRef, Set<Cell> keys);
 
+    /**
+     * Behaves like {@link Transaction#delete}, but additionally stores {@link ChangeMetadata} for the deleted cells.
+     * This metadata is forwarded to TimeLock when acquiring locks at the beginning of the commit protocol.
+     * If two cells in the same row have metadata and the {@link ConflictHandler} for the table acquires row locks,
+     * {@link Transaction#commit} will fail.
+     *
+     * @param tableRef the table from which to delete the values and for which to store metadata
+     * @param keysWithMetadata the cells to delete associated with the metadata that should be stored for
+     * them
+     */
+    @Idempotent
+    void deleteWithMetadata(TableReference tableRef, Map<Cell, ChangeMetadata> keysWithMetadata);
+
```

Comment:
When writing the RFC, I forgot that we also have to add this interface. This is trivial, though as `delete()` is pretty much the same as `put(EMPTY_BYTE_ARRAY)` inernally